controllers/apiserver.go

```diff
@@ -60,14 +72,28 @@ func (r *DSPAReconciler) ReconcileAPIServer(ctx context.Context, dsp *dspav1alph
 	log.Info("Applying APIServer Resources")
 
 	for _, template := range apiServerTemplates {
-		err := r.Apply(dsp, params, template)
+		tmplManifest, err := r.manifest(template, params)
+		if err != nil {
+			return err
+		}
+
+		if template == serverDeployment {
+			r.addComponentName(apiServerComponentName, tmplManifest.Resources()[0].GetName())
```

Comment:
This seems like a lot of work to fetch a Deployment name. Would it not be simpler to specify in the `params` struct 3 new fields: 

```
APIServerDeploymentName
PersistentAgentDeploymentName
ScheduledWorkflowDeploymentName
```

Then utilize these here: [1](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/internal/apiserver/deployment.yaml.tmpl#L4), [2](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/internal/persistence-agent/deployment.yaml.tmpl#L4), [3](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/internal/scheduled-workflow/deployment.yaml.tmpl#L4)

And let params struct be the source of truth for these names, accessible throughout the codebase, as one of the first things we do is generate this struct before doing anything else.

Reply:
Makes total sense @HumairAK.
I pushed the changes. Can you have a second look, please?